lambdas/call-dcmaw-async-cri/src/test/java/uk/gov/di/ipv/core/calldcmawasynccri/pact/dcmawasynccri/ContractTest.java

```diff
@@ -0,0 +1,184 @@
+        // painful.
+        when(mockKmsEs256SignerFactory.getSigner(any())).thenReturn(mockSigner);
+        when(mockSigner.sign(any(), any()))
+                .thenReturn(
+                        new Base64URL(
+                                "UJZPEyLebOJJzyhgkCXL86M1scVSyOyUVym8icII_S3mogaU2dIZpzX6m6-pLMwXm8iWYfouU-AitQuhKD_TTA"));
+        when(mockSigner.supportedJWSAlgorithms()).thenReturn(Set.of(JWSAlgorithm.ES256));
+        when(mockSecureTokenHelper.generate())
+                .thenReturn("ScnF4dGXthZYXS_5k85ObEoSU04W-H3qa_p6npv2ZUY");
+
+        // We need to generate a fixed request, so we set the secure token and expiry to constant
+        // values.
+        var underTest =
+                new CriApiService(
+                        mockConfigService,
+                        mockKmsEs256SignerFactory,
+                        mockSecureTokenHelper,
+                        CURRENT_TIME);
+
+        // Act
+        BearerAccessToken accessToken =
+                underTest.fetchAccessToken(
+                        getCriCallbackRequest("dummyAuthCode"), CRI_OAUTH_SESSION_ITEM);
+        // Assert
+        assertThat(accessToken.getType(), is(AccessTokenType.BEARER));
+        assertThat(accessToken.getValue(), notNullValue());
+        assertThat(accessToken.getLifetime(), greaterThan(0L));
+    }
+
+    @NotNull
+    private static CriCallbackRequest getCriCallbackRequest(String dummyAuthCode) {
+        return new CriCallbackRequest(
+                dummyAuthCode,
+                DCMAW_ASYNC_CRI,
+                "dummySessionId",
+                "https://identity.staging.account.gov.uk/credential-issuer/callback?id=dcmawAsync",
+                "dummyState",
+                null,
+                null,
+                "dummyIpAddress",
+                "dummyDeviceInformation",
+                List.of("dummyFeatureSet"));
+    }
+
+    @NotNull
+    private static OauthCriConfig getMockCredentialIssuerConfig(MockServer mockServer)
+            throws URISyntaxException {
+        return OauthCriConfig.builder()
+                .tokenUrl(new URI("http://localhost:" + mockServer.getPort() + "/token"))
+                .credentialUrl(
+                        new URI("http://localhost:" + mockServer.getPort() + "/credential/issue"))
+                .authorizeUrl(new URI("http://localhost:" + mockServer.getPort() + "/authorize"))
+                .clientId(IPV_CORE_CLIENT_ID)
+                .signingKey(CRI_SIGNING_PRIVATE_KEY_JWK)
+                .encryptionKey(CRI_RSA_ENCRYPTION_PUBLIC_JWK)
+                .componentId(TEST_ISSUER)
+                .clientCallbackUrl(
+                        URI.create(
+                                "https://identity.staging.account.gov.uk/credential-issuer/callback?id=dcmawAsync"))
+                .requiresApiKey(true)
+                .requiresAdditionalEvidence(false)
+                .build();
+    }
+}
```

Comment:
There isn't a test here for bad username/password when getting the access token, and there are no tests for the credential endpoint